apps/settings/js/panels/call/panel.js

```diff
@@ -0,0 +1,472 @@
+        });
+      },
+
+      _initCallWaiting: function() {
+        this._elements.alertPanelSetButton.onclick = () => {
+          var checked = this._elements.alertPanelConfirmInput.checked;
+          this._conn.setCallWaitingOption(checked).then(() => {
+            this._elements.alertPanel.hidden = true;
+            this._updateCallWaitingItemState();
+          }, () => {
+            this._elements.alertPanel.hidden = true;
+            this._updateCallWaitingItemState();
+          });
+        };
+
+        // Bind call waiting setting to the input
+        this._elements.callWaitingInput.onchange = () => {
+          var checked = this._elements.callWaitingInput.checked;
+          this._conn.setCallWaitingOption(checked).then(() => {
+            this._updateCallWaitingItemState();
+          }, () => {
+            this._updateCallWaitingItemState();
+          });
+        };
+      },
+
+      _initVoiceMailSettings: function() {
+        // update all voice numbers if necessary
+        SettingsCache.getSettings((results) => {
+          var settings = navigator.mozSettings;
+          var voicemail = navigator.mozVoicemail;
+          var updateVMNumber = false;
+          var numbers = results['ril.iccInfo.mbdn'] || [];
+
+          [].forEach.call(this._conns, (conn, index) => {
+            var number = numbers[index];
+            // If the voicemail number has not been stored into the database yet
+            // we check whether the number is provided by the mozVoicemail API.
+            // In that case we store it into the setting database.
+            if (!number && voicemail) {
+              number = voicemail.getNumber(index);
+              if (number) {
+                updateVMNumber = true;
+                numbers[index] = number;
+              }
+            }
+          });
+
+          if (updateVMNumber) {
+            settings.createLock().set({
+              'ril.iccInfo.mbdn': numbers
+            }).then(() => {
+              this._updateVoiceMailItemState();
+              settings.addObserver('ril.iccInfo.mbdn', () => {
+                this._updateVoiceMailItemState();
+              });
+            });
+          } else {
+            this._updateVoiceMailItemState();
+            settings.addObserver('ril.iccInfo.mbdn', () => {
+              this._updateVoiceMailItemState();
+            });
+          }
+        });
+      },
+
+      _initFDNItem: function() {
+        var iccId = navigator.mozMobileConnections[this._cardIndex].iccId;
+        if (iccId) {
+          var icc = navigator.mozIccManager.getIccById(iccId);
+          if (icc) {
+            icc.getServiceState('fdn').then((hasFdn) => {
+              this._elements.fdnItem.hidden = !hasFdn;
+            });
+          }
+        }
+      },
+
+      _initVoicemailClickEvent: function() {
+        this._elements.voiceMailItem.onclick = () => {
+          DialogService.show('call-voiceMailSettings');
```

Comment:
I would suggest to pass the card index to the dialog instead of depending on `DsdsSettings`.